rs/sns/governance/src/governance.rs

```diff
@@ -256,6 +256,141 @@ impl GovernanceProto {
         function_followee_index
     }
 
+    /// This is analogous to the legacy `build_function_followee_index` function, but for topic
+    /// following.
+    pub fn build_following_index(
+        &self,
+        neurons: &BTreeMap<String, Neuron>,
+    ) -> BTreeMap<Topic, BTreeMap<String, BTreeSet<NeuronId>>> {
+        let mut function_followee_index = BTreeMap::new();
+        for neuron in neurons.values() {
+            GovernanceProto::add_neuron_to_following_index(&mut function_followee_index, neuron);
+        }
+        function_followee_index
+    }
+
+    /// This is analogous to the legacy `add_neuron_to_function_followee_index` function, but for
+    /// topic following.
+    pub fn add_neuron_to_following_index(
+        index: &mut BTreeMap<Topic, BTreeMap<String, BTreeSet<NeuronId>>>,
+        neuron: &Neuron,
+    ) {
+        let Some(follower_id) = &neuron.id else {
+            log!(ERROR, "Neuron {:?} does not have an ID!", neuron);
+            return;
+        };
+
+        let Some(topic_followees) = &neuron.topic_followees else {
+            return;
+        };
+
+        for (topic, FolloweesForTopic { followees, .. }) in &topic_followees.topic_id_to_followees {
+            let Ok(topic) = Topic::try_from(*topic) else {
+                log!(
+                    ERROR,
+                    "Neuron {} has followees for an invalid topic ID: {}",
+                    follower_id,
+                    topic
+                );
+                continue;
+            };
+            let topoc_index = index.entry(topic).or_default();
+
+            for Followee {
+                neuron_id: followee_id,
+                alias,
+            } in followees
+            {
+                let Some(followee_id) = followee_id else {
+                    let alias = alias
+                        .as_ref()
+                        .map(|alias| format!(" ({})", alias))
+                        .unwrap_or_default();
+                    log!(
+                        ERROR,
+                        "Neuron with ID {:?} has a followee{} with no ID!",
+                        follower_id,
+                        alias
+                    );
+                    continue;
+                };
+
+                let key = followee_id.to_string();
+                topoc_index
+                    .entry(key)
+                    .or_default()
+                    .insert(follower_id.clone());
+            }
+        }
+    }
+
+    /// This is analogous to the legacy `remove_neuron_from_function_followee_index` function, but for
+    /// topic following.
+    pub fn remove_neuron_from_following_index(
```

Comment:
NC

IIUC, this would be used when a neuron changes its following. In that case, `neuron` would be the old value, right?

Anyway, if updating is the use case for this, maybe instead of providing remove, we should instead provide (approximately!)`update(&old_neuron, &new_neuron, &mut index)` ?

Reply:
Note that for now, I'm just following the exact pattern that was there before for the legacy following index. I suppose refactoring them would need to happen side-by-side, so let's consider it separately from this PR.